src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateRepresentationService.java

```diff
@@ -0,0 +1,159 @@
+            .name(solicitorToAdd.getFullName())
+            .email(solicitorToAdd.getEmail())
+            .organisation(change.getOrganisationToAdd())
+            .build();
+
+        ChangedRepresentative removedSolicitor = Optional.ofNullable(change.getOrganisationToRemove())
+            .map(org -> ChangedRepresentative.builder()
+            .name(isApplicant ? getApplicantSolicitorName(caseDetails)
+                : (String) caseDetails.getData().get(RESP_SOLICITOR_NAME))
+            .email(isApplicant ? getApplicantSolicitorEmail(caseDetails)
+                : (String) caseDetails.getData().get(RESP_SOLICITOR_EMAIL))
+            .organisation(org).build())
+            .orElse(null);
+
+        log.info("About to start updating solicitor details in the case data for caseId: {}", caseDetails.getId());
+        caseDetails.getData().putAll(updateCaseDataWithNewSolDetails(caseDetails, addedSolicitor));
+
+        return updateChangeOfRepresentatives(caseDetails, addedSolicitor, removedSolicitor);
+    }
+
+    private Map<String, Object> updateChangeOfRepresentatives(CaseDetails caseDetails,
+                                                              ChangedRepresentative addedSolicitor,
+                                                              ChangedRepresentative removedSolicitor) {
+
+        Map<String, Object> caseData = caseDetails.getData();
+        ChangeOfRepresentatives current = ChangeOfRepresentatives.builder()
+            .changeOfRepresentation(objectMapper.convertValue(caseData.get(CHANGE_OF_REPS),
+                new TypeReference<>() {}))
+            .build();
+
+        ChangeOfRepresentatives change = changeOfRepresentationService.generateChangeOfRepresentatives(
+            ChangeOfRepresentationRequest.builder()
+                .by(addedSolicitor.getName())
+                .party(isApplicant ? "Applicant" : "Respondent")
+                .clientName(isApplicant ? caseDataService.buildFullApplicantName(caseDetails)
+                    : caseDataService.buildFullRespondentName(caseDetails))
+                .current(current)
+                .addedRepresentative(addedSolicitor)
+                .removedRepresentative(removedSolicitor)
+                .build()
+        );
+
+        caseData.put(CHANGE_OF_REPS, change.getChangeOfRepresentation());
+
+        return caseData;
+    }
+
+    private Map<String, Object> updateCaseDataWithNewSolDetails(CaseDetails caseDetails,
+                                                                ChangedRepresentative addedSolicitor) {
+
+        Map<String, Object> caseData = caseDetails.getData();
+        boolean isConsented = caseDataService.isConsentedApplication(caseDetails);
+        String appSolicitorAddressField = isConsented ? CONSENTED_SOLICITOR_ADDRESS : CONTESTED_SOLICITOR_ADDRESS;
+        String solicitorAddressField = isApplicant ? appSolicitorAddressField : RESP_SOLICITOR_ADDRESS;
+
+        //isRepresented Boolean
```

Comment:
comments are usually not necessary, the code should be self-explanatory. In fact, checkstyle should have blocked this.